tools/webpack-plugin/src/helpers/statsPrinter.ts

```diff
@@ -0,0 +1,52 @@
+import webpack from 'webpack';
+import { AssetStats } from '../models/AssetStats';
+
+function statToString(stat: boolean | string | Error) {
+    if (typeof stat === 'string') {
+        return stat;
+    }
+
+    if (typeof stat === 'boolean') {
+        return stat ? 'successful' : 'skipped';
+    }
+
+    return stat.message;
+}
+
+export function statsPrinter(logger: webpack.Compilation['logger']) {
```

Comment:
Let's think about similar use cases. When I think about what we need to do I think its fair to compare it with an output from for example jest. Why? During jest tests all the time you see incremental output about passed/failed tests. 

The user might stop them any time. The user in the end sees how many tests passed/failed. In the end when we want to see the summary.

Why we don't collect information about successes/failures and print information in real time. Uploading symbols consumes space. Maybe it's not the case right now, however, it will be a case in the future. The user wants to stop the uploading process asap when the user sees an issue.  

tl;dr;
* compare the output with jest - we spam too much redundant information in the end of the cli - for projects like workflows and unexpected error (let's say type error in our code somewhere) you will generate 1300 error messages in the end as a user I expect to see stats. 
* we can show errors/useful information in real-time to the user during processing and in the end just show stats.  The user already has everything needed to understand where is an error.
* I love to code, but we can avoid doing that just for simply incrementing success/failure variables

Reply:
These are Webpack guidelines. Webpack does not spam anything during the build process, and only prints stats at the end. If you enable more verbose logging, you will have compilation-time logs.

See https://webpack.js.org/api/logging/.

![image](https://github.com/backtrace-labs/backtrace-javascript/assets/9085864/7c5b6d71-f80a-4cf6-8bc1-4d32f729261d)

>you will generate 1300 error messages in the end as a user I expect to see stats.
If you output 1300 files through Webpack, something is definitely wrong with your setup. Still, Webpack's output will be 30x larger in this case than what we output.

The most I could do is not to print each file separately in default log mode, and display just a status of what has been done for all files.